### PR TITLE
[shaderwriter] Update to version 2.3.0.

### LINKED ISSUE #25131: [shaderwriter] update to 2.3.0.
Library name: shaderwriter

New version number: 2.3.0

### PR DESCRIPTION
**Pull request by DragonJoker**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #25131 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All.
  No.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Hopefully yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by JackBoosY** [comment]
I found two thing in the genreated file list:
1.
```
shaderwriter:x64-linux:/share/shaderwriter/SdwCompilersConfig-debug.cmake
shaderwriter:x64-linux:/share/shaderwriter/SdwCompilersConfig-release.cmake
shaderwriter:x64-linux:/share/shaderwriter/SdwCompilersConfig.cmake
```
These cmake files should be installed into _share/SdwCompilers_.

2.
```
shaderwriter:x64-windows:/bin/sdwCompilerGlsl.dll
shaderwriter:x64-windows:/bin/sdwCompilerHlsl.dll
shaderwriter:x64-windows:/bin/sdwCompilerSpirV.dll
shaderwriter:x64-windows:/bin/sdwShaderAST.dll
shaderwriter:x64-windows:/bin/sdwShaderWriter.dll
shaderwriter:x64-windows:/debug/bin/sdwCompilerGlsld.dll
shaderwriter:x64-windows:/debug/bin/sdwCompilerHlsld.dll
shaderwriter:x64-windows:/debug/bin/sdwCompilerSpirVd.dll
shaderwriter:x64-windows:/debug/bin/sdwShaderASTd.dll
shaderwriter:x64-windows:/debug/bin/sdwShaderWriterd.dll
```
There is no pdbs, please add `vcpkg_copy_pdbs()` after `vcpkg_cmake_configure`.

File: ports/shaderwriter/portfile.cmake [hunk 1]
```diff
@@ -29,6 +29,7 @@ vcpkg_cmake_configure(
         -DSDW_UNITY_BUILD=ON
 )
 
+vcpkg_copy_pdbs()
 vcpkg_cmake_install()
 vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/shaderwriter)
```
Comment (JackBoosY) [inline ports/shaderwriter/portfile.cmake:34]: For the first suggestion:
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME ShaderWriter CONFIG_PATH lib/cmake/shaderwriter DO_NOT_DELETE_PARENT_CONFIG_PATH)
vcpkg_cmake_config_fixup(PACKAGE_NAME SdwCompilers CONFIG_PATH lib/cmake/shaderwriter)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/shaderwriter/portfile.cmake (hunk 1 above)
Comment (DragonJoker) [reply ports/shaderwriter/portfile.cmake:34]: Well, I've actually reworked the install script in the project, hence this change won't be needed.

Comment (PhoebeHui) [inline ports/shaderwriter/portfile.cmake:34]:
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME ShaderWriter CONFIG_PATH lib/cmake/shaderwriter)
```

Comment (PhoebeHui) [reply ports/shaderwriter/portfile.cmake:34]: See https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/ports/vcpkg-cmake-config/vcpkg_cmake_config_fixup.md.

Comment (DragonJoker) [reply ports/shaderwriter/portfile.cmake]: Done.

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the update!

**Pull request merged**